source/common/network/connection_impl.cc

```diff
@@ -395,6 +396,21 @@ void ConnectionImpl::updateWriteBufferStats(uint64_t num_written, uint64_t new_s
                                            buffer_stats_->write_current_);
 }
 
+const std::string Network::ConnectionImpl::destinationAddress() {
+  if (fd_ != -1) {
+    sockaddr_storage orig_dst_addr;
+    memset(&orig_dst_addr, 0, sizeof(orig_dst_addr));
+    bool success = Utility::getOriginalDst(fd_, &orig_dst_addr);
+    if (success) {
+      return Utility::urlForTcp(
+          Utility::getAddressName(reinterpret_cast<struct sockaddr_in*>(&orig_dst_addr)),
+          Utility::getAddressPort(reinterpret_cast<struct sockaddr_in*>(&orig_dst_addr)));
+    }
+  }
+
+  return EMPTY_STRING;
```

Comment:
Per my previous interface comment, can we actually add a member variable called local_address_, and populate in the constructor correctly. The way to do this would be to have ConnectionImpl() constructor take a local_address also as a param. This will plumb through quite nicely, since now you can populate the address directly in the listener.

NOTE: In the interest of time, I'm OK with it if it's currently the empty string for client connections, and even server connection that don't come in via getOrginalDst. We can add better support later for these things.